### PR TITLE
Use pseudo-assembly fallback in RKI output, QC overview table and …

### PR DESCRIPTION
**Pull request by thomasbtf**
…pangolin calls

### CODE REVIEW

File: workflow/rules/common.smk [hunk 1]
```diff
@@ -459,6 +461,76 @@ def get_adapters(wildcards):
     return config["adapters"]["illumina-revelo"]
 
 
+def get_final_assemblies(wildcards):
+    if wildcards.assembly_type == "masked-assembly":
+        pattern = "results/{{date}}/contigs-masked/{sample}.fasta"
+    elif wildcards.assembly_type == "pseudo-assembly":
+        pattern = "results/{{date}}/contigs-masked/{sample}.fasta"
```
Comment (alethomas) [inline workflow/rules/common.smk:468]: Is it on purpose to use the same path here?

File: workflow/rules/common.smk [hunk 2]
```diff
@@ -459,6 +461,76 @@ def get_adapters(wildcards):
...
+    ).output[0].open() as f:
+        masked_samples = pd.read_csv(f, squeeze=True).astype(str).to_list()
+
+    with checkpoints.rki_filter.get(
+        date=wildcards.date, assembly_type="pseudo-assembly"
+    ).output[0].open() as f:
+        pseudo_samples = pd.read_csv(f, squeeze=True).astype(str).to_list()
+
+    pseudo_assembly_pattern = "results/{{date}}/pseudoassembled-contigs/{sample}.fasta"
+    normal_assembly_pattern = "results/{{date}}/contigs-masked/{sample}.fasta"
```
Comment (alethomas) [inline workflow/rules/common.smk:494]: When I tried to solve the problem I thought of renaming all the different contigs from e.g. /peudoassambled-contigs/ /contigs/pseudo/, which would make the naming a little more consistent

**Review by alethomas — APPROVED** [review]
Very nice solution

File: workflow/rules/common.smk (hunk 1 above)
Comment (thomasbtf) [reply workflow/rules/common.smk]: Nah, that was a copy-paste mistake. Fixed in 874adcf.

File: workflow/rules/common.smk (hunk 2 above)
Comment (thomasbtf) [reply workflow/rules/common.smk]: Good idea! Improved in 874adcf.

File: workflow/rules/common.smk [hunk 3]
```diff
@@ -459,6 +462,82 @@ def get_adapters(wildcards):
...
+        ).output[0].open() as f:
+            pseudo_samples = pd.read_csv(f, squeeze=True, header=None).astype(str).to_list()
+    # for testing of pangolin don't create pseudo-assembly
+    else:
+        masked_samples = [wildcards.sample]
+
+    pseudo_assembly_pattern = "results/{{date}}/contigs/pseudoassembled/{sample}.fasta"
+    normal_assembly_pattern = "results/{{date}}/contigs/masked/{sample}.fasta"
+
+    # get accepted sampels for rki submission
```
Comment (johanneskoester) [inline workflow/rules/common.smk:501]:
```suggestion
    # get accepted samples for rki submission
```

File: workflow/rules/common.smk [hunk 4]
```diff
@@ -459,6 +462,82 @@ def get_adapters(wildcards):
...
+                )
+            else:
+                accepted_assemblies.append(
+                    pseudo_assembly_pattern.format(sample=sample)
+                )
+            print(accepted_assemblies)
+        return accepted_assemblies
+
+    # for the pangolin call
+    if agg_typ == "single sample":
```
Comment (johanneskoester) [inline workflow/rules/common.smk:519]:
```suggestion
    elif agg_typ == "single sample":
```

File: workflow/rules/common.smk [hunk 5]
```diff
@@ -459,6 +462,82 @@ def get_adapters(wildcards):
...
+            pseudo_samples = pd.read_csv(f, squeeze=True, header=None).astype(str).to_list()
+    # for testing of pangolin don't create pseudo-assembly
+    else:
+        masked_samples = [wildcards.sample]
+
+    pseudo_assembly_pattern = "results/{{date}}/contigs/pseudoassembled/{sample}.fasta"
+    normal_assembly_pattern = "results/{{date}}/contigs/masked/{sample}.fasta"
+
+    # get accepted sampels for rki submission
+    if agg_typ == "accepted samples":
```
Comment (johanneskoester) [inline workflow/rules/common.smk:502]: Variable name should be english: agg_type

File: workflow/rules/common.smk [hunk 6]
```diff
@@ -459,6 +462,82 @@ def get_adapters(wildcards):
...
+        if wildcards.sample in masked_samples:
+            return "results/{date}/contigs/polished/{sample}.fasta"
+        elif wildcards.sample in pseudo_samples:
+            return "results/{date}/contigs/pseudoassembled/{sample}.fasta"
+        # for not accepted samples use the polished-contigs
+        else:
+            return "results/{date}/contigs/polished/{sample}.fasta"
+
+    # for the qc report
+    if agg_typ == "all samples":
```
Comment (johanneskoester) [inline workflow/rules/common.smk:529]:
```suggestion
    elif agg_typ == "all samples":
```

File: workflow/rules/generate_output.smk [hunk 7]
```diff
@@ -129,7 +115,7 @@ rule virologist_report:
         ),
         initial_contigs=lambda wildcards: get_expanded_contigs(wildcards),
         polished_contigs=lambda wildcards: expand(
-            "results/{{date}}/polished-contigs/{sample}.fasta",
+            "results/{{date}}/contigs/polished/{sample}.fasta",
```
Comment (johanneskoester) [inline workflow/rules/generate_output.smk:118]: Mhm, here I am confused. shouldn't this either obtain the polished (or better even masked) or the pseudoassembled contigs, depending on the rki filtering?

**Review by johanneskoester — CHANGES_REQUESTED** [review]

File: workflow/rules/common.smk (hunk 5 above)
Comment (thomasbtf) [reply workflow/rules/common.smk]: Thanks for spotting this! Addressed in a15f831.

File: workflow/rules/generate_output.smk (hunk 7 above)
Comment (thomasbtf) [reply workflow/rules/generate_output.smk:118]: I actually wasn't sure, how we should handle this. The current use of this input is the calculation of the `Final Contig (bp)` length, thus it provides information about the "normal assembly". I didn't want to skew our view on this data by mixing it with information from the pseudo assembly. 

I thought about adding another column with the length of the pseudo assembly. But what is the added value for the virologist, when would do this?

However, I would love to hear your opinions about this.

Comment (johanneskoester) [reply workflow/rules/generate_output.smk:118]: Ah, I see. I think it is best to have one column for the assembly length, one for the pseudoassembly length and one denoting which assembly was chosen.

Comment (thomasbtf) [reply workflow/rules/generate_output.smk:118]: Added this in 009a33a :)

**Review by johanneskoester — APPROVED** [review]

**Pull request merged**